src/server.py

```diff
@@ -399,19 +401,41 @@ def play_sound(self, sound_event_id, connection, sender, path, iface,
             metadata = self.metadata[sound_event_id]
             self.players[uuid_] = HackSoundPlayer(uuid_, metadata, sender,
                                                   options)
+            self._watch_bus_name(sender, uuid_)
             # Insert the uuid in the dictionary organized by sound event id.
             if overlap_behavior != "overlap":
                 self._uuid_by_event_id[sound_event_id] = uuid_
 
-            self.players[uuid_].connect("eos", self.__player_eos_cb,
-                                        sound_event_id, uuid_)
+            self.players[uuid_].connect("eos", self.__player_eos_cb, sender,
+                                        sound_event_id, uuid_,)
             self.players[uuid_].connect("error", self.__player_error_cb,
-                                        sound_event_id, uuid_,
+                                        sender, sound_event_id, uuid_,
                                         connection, path, iface)
             self.players[uuid_].play()
 
         return invocation.return_value(GLib.Variant('(s)', (uuid_, )))
 
+    def _watch_bus_name(self, bus_name, uuid_):
+        if bus_name not in self._uuid_by_bus_name:
```

Comment:
Having a single data structure will also make this method more readable. Something like:
```
{
  'bus_name_A': { 'watcher_id': watcher_id, 'uuids': set(...) },
  'bus_name_B': {...},
}
```

Reply:
I was thinking something like this. What about using a NamedTuple?
WatcherInfo = namedtuple("WatcherInfo", ["watcher_id", "uuids"], verbose=True)
instead of 
{ 'watcher_id': watcher_id, 'uuids': set(...) }
?